src/main.ts

```diff
@@ -0,0 +1,186 @@
+import { Euler, Quaternion, Vector3, MathUtils } from 'three'
+import { EPSILON } from './constants'
+
+const xAxis = new Quaternion(-1, 0, 0, 0)
+const zAxis = new Quaternion(0, 0, +1, 0)
+
+const quatA = new Quaternion()
+const quatB = new Quaternion()
+const quatC = new Quaternion()
+const quatD = new Quaternion()
+const quatE = new Quaternion()
+
+const vecA = new Vector3()
+const vecB = new Vector3()
+const vecC = new Vector3()
+const vecD = new Vector3()
+const vecE = new Vector3()
+const vecF = new Vector3()
+const vecG = new Vector3()
+const vecH = new Vector3()
+
+// Golang: https://github.com/viamrobotics/rdk/blob/main/spatialmath/orientationVector.go
+// Rust:   https://github.com/viamrobotics/rust-utils/blob/main/src/spatialmath/utils.rs
+
+/**
+ * Viam’s orientation vector is a method for describing the orientation of an object in 3D space.
+ * It is part of a Pose which also includes the position in 3D space.
+ * 
+ * The vector extends from the center of the object to another point in the reference frame. This defines the direction something is pointing in.
+ */
+export class OrientationVector {
+  readonly isOrientationVector = true
+
+  #vec = new Vector3()
+
+  /**
+   * The vector's x component.
+   * @default 0
+   */
+  set x (value: number) {
+    this.#vec.setX(value).normalize()
+  }
+
+  get x (): number {
+    return this.#vec.x
+  }
+
+  /**
+   * The vector's y component.
+   * @default 0
+   */
+  set y (value: number) {
+    this.#vec.setY(value).normalize()
```

Comment:
Should we really be normalizing on the setting of individual fields?

If I'm a user and I start with an OV pointing at xyz {0,0,1}, then set y to 1, then set x to 1, I would expect the vector to point at the normalized {1,1,1}, rather than normalized {1,0.5,0.5}, which is what this would do as far as I can tell.

On the other hand- I don't know Javascript at all, so I might be misreading this.